internal/codespaces/connection/connection.go

```diff
@@ -74,6 +81,44 @@ func NewCodespaceConnection(ctx context.Context, codespace *api.Codespace, httpC
 	}, nil
 }
 
+// Connect connects the client to the tunnel.
+func (c *CodespaceConnection) Connect(ctx context.Context) error {
+	// Lock the mutex to prevent connection races
+	c.TunnelClient.connectMu.Lock()
+
+	// If already connected, return
+	if c.TunnelClient.connected {
+		return nil
+	}
+
+	// Connect to the tunnel
+	if err := c.TunnelClient.Client.Connect(ctx, ""); err != nil {
+		return fmt.Errorf("error connecting to tunnel: %w", err)
+	}
+
+	// Set the connected flag so we know we're connected
+	c.TunnelClient.connected = true
+
+	// Unlock the mutex
+	c.TunnelClient.connectMu.Unlock()
```

Comment:
I think you need move this up to a `defer c.TunnelClient.connectMu.Unlock()` right after the `Lock` call otherwise both the `return nil` and `return err` paths will hold the lock forever.

Reply:
Sure, it definitely wouldn't hurt to do that